### ISSUE TITLE
MultiStepLR for multi gamma

### ISSUE THREAD

**Issue by robotcator** [pytorch/pytorch#2548]
Hi, sometimes I may confront with the situation that schedule the learning rate with multi `gamma`. For the current MultiStepLR, the optimizer uses lr for all groups.
For example:
```
>>> # lr = 0.05     if epoch < 30
>>> # lr = 0.005    if 30 <= epoch < 80
>>> # lr = 0.0005   if epoch >= 80
```
But we may have a list `gammas`, such as milestones=[30, 80], gamma=[0.1, 0.2]
```
>>> # lr = 0.05     if epoch < 30
>>> # lr = 0.05*0.1    if 30 <= epoch < 80
>>> # lr = 0.05*0.1*0.2   if epoch >= 80
```
This is my code for this feature:
```
from torch.optim.lr_scheduler import _LRScheduler
class myMultiStepLR(_LRScheduler):
    def __init__(self, optimizer, milestones, gamma, last_epoch=-1):
        if not list(milestones) == sorted(milestones):
            raise ValueError('Milestones should be a list of'
                             ' increasing integers. Got {}', milestones)
        self.milestones = milestones
        self.gamma = gamma
        super(myMultiStepLR, self).__init__(optimizer, last_epoch)

    def get_lr(self):
        if self.last_epoch in self.milestones:
            index = self.milestones.index(self.last_epoch)
            _gamma = self.gamma[index]
        else:
            _gamma = 1.0
        return [base_lr * _gamma for base_lr in self.base_lrs]
```

cc @vincentqb

**Comment by Jiaming-Liu** [comment]
This does not seem very common. LambdaLR is designed for painless customizing.

**Comment by vincentqb** [comment]
You can also use `MultiplicativeLR` for this now. Since there are solutions for this, I will close this issue. Please feel free to re-open if you would like to continue the discussion.
